src/python/pants/option/option_value_container.py

```diff
@@ -23,8 +58,7 @@ class OptionValueContainer:
        See ranked_value.py for more details.
     """
 
-    def __init__(self) -> None:
-        self._value_map: Dict[Key, RankedValue] = {}
+    _value_map: Dict[Key, RankedValue]
```

Comment:
Technically, should be `FrozenDict`, but this would also require that all values are frozen too, which isn't fun code to write.

`__hash__` is working okay on this? I'd expect it to fail.

Reply:
It's not: from the description:
> Give `Subsystem` and `OptionsValueContainer` structural implementations of `__eq__` by making them dataclasses. Because they will only be used as return values / positional args (ie, the rust side `Value` struct), they don't need `__hash__`.

Reply:
Hm, but `frozen=True` generates `__hash__`. I think you need to override `__hash__` to say that it's not implemented.

> If a class that does not override `__eq__()` wishes to suppress hash support, it should include `__hash__ = None` in the class definition.

Reply:
It generates it, but if you call it, it will fail.